components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/policy/PolicyAttributeBuilder.java

```diff
@@ -94,6 +95,22 @@ public Properties getPolicyMetaDataFromPolicy() throws EntitlementException {
         return properties;
     }
 
+    /**
+     * This retrieves attributes from the policy.
+     *
+     * @return attributeDTO list.
+     * @throws EntitlementException if an error occurs while retrieving attributes.
+     */
+    public List<AttributeDTO> getAttributesFromPolicy() throws EntitlementException {
+
+        List<AttributeDTO> attributeDTOs = new ArrayList<>();
+        try {
+            return createPolicyMetaData(policy, attributeDTOs);
+        } catch (EntitlementException e) {
+            throw new EntitlementException("Can not create Policy MetaData for given policy");
```

Comment:
are we catching and throwing same exception? can we include the ```e``` also in the content.